tests/nosetests/pyanaconda_tests/module_payload_dnf_manager_test.py

```diff
@@ -0,0 +1,91 @@
+#
+# Copyright (C) 2020  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import unittest
+from unittest.mock import patch
+
+from pyanaconda.modules.payloads.payload.dnf.dnf_manager import DNFManager
+
+
+class DNFMangerTestCase(unittest.TestCase):
```

Comment:
I don't want to mock `read_comps`. It seems to work fine and the only reason we call it during the initialization is to set up `base.comps`. Otherwise, it would be `None`. Anyway, I plan to remove it from the initialization and handle `None`, but it will take a while.

Also, I am not going to use a mock instead of DNF, because that doesn't test anything. We don't mock Blivet, just the necessary parts. I don't see a difference. This solution already helped me to discover some bugs in our code.